src/lib/ride.ts

```diff
@@ -0,0 +1,39 @@
+import RideModel from '../database/model/ride';
+import {dataBase} from '../database';
+
+export function startRide() {
+  // TODO stop any current rides.
+  return dataBase.write(async () => {
+    return dataBase.get<RideModel>('ride').create(r => {
+      r.startedAt = new Date().getUTCMilliseconds();
+      return r;
+    });
+  });
+}
+
+export function stopRide(ride: RideModel) {
+  return dataBase.write(async () => {
+    return ride.update(() => {
+      ride.endedAt = new Date().getUTCMilliseconds();
```

Comment:
Not sure if its a problem, but i noticed that if you stop a ride will its in the paused state (probably fairly common) it doesn't obviously stays paused. Not sure if maybe we want to say that by default a stopped ride can no longer be paused, and therefore we should set is_paused back to 0 here?

Reply:
Great point, we should definitely do that.